src/pyfmi/fmi.pyx

```diff
@@ -3442,17 +3445,46 @@ cdef class FMUModelME1(FMUModelBase):
     the low-level FMI function: fmiSetContinuousStates/fmiGetContinuousStates.
     """)
 
+
+    cdef int __get_nominal_continuous_states(self, FMIL.fmi1_real_t* xnominal, size_t nx):
+        return FMIL.fmi1_import_get_nominal_continuous_states(self._fmu, xnominal, nx)
+
     def _get_nominal_continuous_states(self):
-        cdef int status
-        cdef N.ndarray[FMIL.fmi1_real_t, ndim=1,mode='c'] ndx = N.zeros(self._nContinuousStates,dtype=N.double)
+        """
+        Returns the nominal values of the continuous states.
 
-        status = FMIL.fmi1_import_get_nominal_continuous_states(
-                self._fmu, <FMIL.fmi1_real_t*>ndx.data, self._nContinuousStates)
+        Returns::
+            The nominal values.
+        """
+        cdef int status
+        cdef N.ndarray[FMIL.fmi1_real_t, ndim=1, mode='c'] xn = N.zeros(self._nContinuousStates, dtype=N.double)
 
+        status = self.__get_nominal_continuous_states(<FMIL.fmi1_real_t*> xn.data, self._nContinuousStates)
         if status != 0:
             raise FMUException('Failed to get the nominal values.')
 
-        return ndx
+        # Fallback - auto-correct the illegal nominal values:
+        xvrs = self.get_state_value_references()
+        # TODO: What about xvrs[i] == fmiUndefinedValueReference?
```

Comment:
Something we need to care about? I've never seen this before

Reply:
Can this happen? If it can I suggest that we check against it and return "state variable <index>" instead. I also have not seen this.

Reply:
It's removed in FMI2 and after looking at it a bit more I don't even understand how it would work in the first place. It's defined in the C header as unsigned -1, which is platform and compiler dependent on the size of unsigned int. 

It's probably very unlikely that we need to check for it, given that most tools want to compile for both ME1 and ME2, and it doesn't make sense to have special behavior for FMI1.

We'd also get hit by it in other places.

I suggest just removing the TODO.

Reply:
(I removed the TODO already, but thread is still open for discussion)